pkg/network/bridge/lister.go

```diff
@@ -1,40 +1,88 @@
 package bridge
 
 import (
-	"os"
 	"strings"
 
 	"github.com/golang/glog"
 	"github.com/kubevirt/device-plugin-manager/pkg/dpm"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
 )
 
 const (
 	resourceNamespace              = "bridge.network.kubevirt.io"
 	BridgesListEnvironmentVariable = "BRIDGES"
 	// maximal interface name length (15) - nic index suffix (3)
 	maxBridgeNameLength = 12
+	configMapName       = "device-plugin-network-bridge"
 )
 
+func getBridgeList(configMap *v1.ConfigMap) dpm.PluginNameList {
+	var plugins = make(dpm.PluginNameList, 0)
+
+	bridges := strings.Split(configMap.Data["bridges"], ",")
+
+	for _, bridgeName := range bridges {
+		if len(bridgeName) > maxBridgeNameLength {
+			glog.Fatalf("Bridge name (%s) cannot be longer than %d characters", bridgeName, maxBridgeNameLength)
+		}
+		plugins = append(plugins, bridgeName)
+	}
+
+	return plugins
+}
+
 type BridgeLister struct{}
 
 func (bl BridgeLister) GetResourceNamespace() string {
 	return resourceNamespace
 }
 
 func (bl BridgeLister) Discover(pluginListCh chan dpm.PluginNameList) {
-	var plugins = make(dpm.PluginNameList, 0)
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		glog.Fatalf(err.Error())
+	}
 
-	bridgesListRaw := os.Getenv(BridgesListEnvironmentVariable)
-	bridges := strings.Split(bridgesListRaw, ",")
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf(err.Error())
 
-	for _, bridgeName := range bridges {
-		if len(bridgeName) > maxBridgeNameLength {
-			glog.Fatalf("Bridge name (%s) cannot be longer than %d characters", bridgeName, maxBridgeNameLength)
-		}
-		plugins = append(plugins, bridgeName)
 	}
 
-	pluginListCh <- plugins
+	watchlist := cache.NewListWatchFromClient(clientset.CoreV1().RESTClient(), string(v1.ResourceConfigMaps), v1.NamespaceAll, fields.Everything())
```

Comment:
This fails with:

```
I0814 12:13:58.790363    2450 reflector.go:240] Listing and watching *v1.ConfigMap from github.com/kubevirt/kubernetes-device-plugins/pkg/network/bridge/lister.go:85
E0814 12:13:58.791881    2450 reflector.go:205] github.com/kubevirt/kubernetes-device-plugins/pkg/network/bridge/lister.go:85: Failed to list *v1.ConfigMap: configmaps is forbidden: User "system:serviceaccount:kube-system:default" cannot list configmaps at the cluster scope
```

We should list only the kube-system namespace, not NamespaceAll.